src/status_im2/common/toasts/events.cljs

```diff
@@ -0,0 +1,50 @@
+(ns status-im2.common.toasts.events
+  (:require
+   [status-im.utils.fx :as fx]
+   [taoensso.encore    :as enc]))
+
+(def ^:private next-toast-id (atom 0))
+
+(fx/defn upsert
+  {:events [:toasts/upsert]}
+  [{:keys [db]} id opts]
+  (let [{:toasts/keys [index toasts]} db
+        update?                       (some #{id} index)
+        index                         (enc/conj-when index (and (not update?) id))
+        toasts                        (assoc toasts id (assoc opts :update? update?))]
+    (enc/assoc-when
+     {:db (-> db
```

Comment:
better to use core clojure function like  `cond->`

Reply:
done